components/movementsensor/rtkutils/cachedData.go

```diff
@@ -174,3 +180,49 @@ func (g *CachedData) Properties(
 		CompassHeadingSupported: true,
 	}, nil
 }
+
+// findDistance calculates the distance between two points on Earth.
+// lat1, lon1: Latitude and Longitude of point 1 (in decimal degrees)
+// lat2, lon2: Latitude and Longitude of point 2 (in decimal degrees)
+// returns the distance in meters.
+func findDistance(lat1, lon1, lat2, lon2 float64) float64 {
+	lat1Rad, lon1Rad := utils.DegToRad(lat1), utils.DegToRad(lon1)
+	lat2Rad, lon2Rad := utils.DegToRad(lat2), utils.DegToRad(lon2)
+
+	dLat := lat2Rad - lat1Rad
+	dLon := lon2Rad - lon1Rad
+
+	a := math.Sin(dLat/2)*math.Sin(dLat/2) +
+		math.Cos(lat1Rad)*math.Cos(lat2Rad)*
+			math.Sin(dLon/2)*math.Sin(dLon/2)
+	c := 2 * math.Atan2(math.Sqrt(a), math.Sqrt(1-a))
+
+	distanceKm := earthRadiusKm * c
+	return distanceKm * 1000 // convert km to meters
+}
+
+// calculateCompassDegreeError calculates the compass degree error
+// of two geo points.
+func (g *CachedData) calculateCompassDegreeError() float64 {
+	firstPos := g.lastPosition.GetLastPosition()
+	secondPos := g.nmeaData.Location
+
+	// Check if the two positions are the same.
+	if firstPos.Lat() == secondPos.Lat() && firstPos.Lng() == secondPos.Lng() {
+		return math.NaN()
+	}
+
+	adjacent := findDistance(firstPos.Lat(), firstPos.Lng(), secondPos.Lat(), secondPos.Lng())
+	// If adjacent is 0, atan2 will be 90 degrees which is not desired.
+	if adjacent == 0 {
+		return math.NaN()
+	}
+	radius := 5.0
+	if g.nmeaData.FixQuality >= 4 {
+		radius = 0.1
```

Comment:
PR description doesn't match code: this looks like the radius is either 5 centimeters or 1 _millimeter._ What are the actual units on the radius?

...come to think of it, what circle does this radius go to? I don't really get why this function does what it does, and could use some more comments somewhere in here.

Reply:
:exploding_head: ...actually, `adjacent` has units of meters, so this is saying the radius is either 5 meters or 10 centimeters! Give this part another look.

Reply:
yeah, I just noticed it too. Good catch! thanks!

Reply:
Updated the description. And adding comments for clarification.